### PR TITLE
Enable Stackinator's --no-bwrap option in CI (for Eiger)

### PR DESCRIPTION
**Pull request by RMeli**
https://github.com/eth-cscs/stackinator/pull/183 added a `--no-bwrap` option to the Stackinator. This option is now needed to build on Eiger.

This PR adds this option, which can be enabled as an (optional) `no_bwrap` field for each cluster.

This PR also fixes the usage of the `--develop` option (see #55).

### CODE REVIEW

File: ci/config/configuration.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def name(self):
     @property
     def spack_develop(self):
         return self._use_spack_develop
-
+
```
Comment (RMeli) [inline ci/config/configuration.py:46]:
```suggestion

```

File: ci/config/configuration.py [hunk 2]
```diff
@@ -143,7 +143,7 @@ def __init__(self, config_path, recipe_path):
 
         if not valid:
             raise ConfigError("configuration error - see log")
-
+
```
Comment (RMeli) [inline ci/config/configuration.py:146]:
```suggestion

```

File: ci/config/configuration.py [hunk 3]
```diff
@@ -183,7 +183,7 @@ def uenvs(self):
     @property
     def clusters(self):
         return self._clusters
-
+
```
Comment (RMeli) [inline ci/config/configuration.py:186]:
```suggestion

```

File: stack-build [hunk 4]
```diff
@@ -53,19 +53,21 @@ build_root="-"
...
 
-while getopts n:s:r:b:u:m:d: flag
+while getopts n:s:r:b:u:m:dw flag
 do
     case "${flag}" in
         n) name=${OPTARG};;
         s) system=${OPTARG};;
         r) recipe_path=$(realpath "${OPTARG}");;
-        b) build_root="${OPTARG}";;
+        b) build_root=$(realpath "${OPTARG}");;
```
Comment (RMeli) [inline stack-build:66]:
```suggestion
        b) build_root="${OPTARG}";;
```
Unrelated change.

**Comment by RMeli** [comment]
cscs-ci run alps;system=eiger;uarch=zen2;uenv=openfoam-org:8

**Comment by RMeli** [comment]
cscs-ci run alps;system=eiger;uarch=zen2;uenv=openfoam-org:8

**Comment by RMeli** [comment]
cscs-ci run alps;system=eiger;uarch=zen2;uenv=openfoam-org:8

**Comment by RMeli** [comment]
cscs-ci run alps;system=eiger;uarch=zen2;uenv=openfoam-org:8

**Review by bcumming — APPROVED** [review]
looks good - thanks!
